arch/risc-v/src/common/riscv_createstack.c

```diff
@@ -218,7 +218,7 @@ void riscv_stack_color(void *stackbase, size_t nbytes)
   uint32_t *stkptr = (uint32_t *)(((uintptr_t)stackbase + 3) & ~3);
   uintptr_t stkend = nbytes ? (((uintptr_t)stackbase + nbytes) & ~3):
                      up_getsp(); /* 0: colorize the running stack */
-  size_t    nwords = (stkend - (uintptr_t)stackbase) >> 2;
+  size_t nwords = (stkend - (uintptr_t)(nbytes ? stackbase : stkptr)) >> 2;
```

Comment:
what about other archs? Do they have the same issue?

Reply:
They don't have this issue (xtensa/arm).